StabilityMatrix.Core/Models/Packages/FluxGym.cs

```diff
@@ -0,0 +1,178 @@
+        InstallPackageOptions options,
+        IProgress<ProgressReport>? progress = null,
+        Action<ProcessOutput>? onConsoleOutput = null,
+        CancellationToken cancellationToken = default
+    )
+    {
+        progress?.Report(new ProgressReport(-1f, "Cloning sd-scripts", isIndeterminate: true));
+        await prerequisiteHelper
+            .RunGit(
+                ["clone", "-b", "sd3", "https://github.com/kohya-ss/sd-scripts"],
+                onConsoleOutput,
+                installLocation
+            )
+            .ConfigureAwait(false);
+
+        progress?.Report(new ProgressReport(-1f, "Setting up venv", isIndeterminate: true));
+        await using var venvRunner = await SetupVenvPure(installLocation).ConfigureAwait(false);
+
+        progress?.Report(
+            new ProgressReport(-1f, "Installing sd-scripts requirements", isIndeterminate: true)
+        );
+        var sdsRequirements = new FilePath(installLocation, "sd-scripts", "requirements.txt");
+        var sdsPipArgs = new PipInstallArgs()
+            .WithParsedFromRequirementsTxt(
+                await sdsRequirements.ReadAllTextAsync(cancellationToken).ConfigureAwait(false),
+                "torch"
+            )
+            .RemovePipArgKey("-e .");
+        await venvRunner.PipInstall(sdsPipArgs, onConsoleOutput).ConfigureAwait(false);
+
+        progress?.Report(new ProgressReport(-1f, "Installing requirements", isIndeterminate: true));
+        var requirements = new FilePath(installLocation, "requirements.txt");
+        var pipArgs = new PipInstallArgs()
+            .AddArg("--pre")
+            .WithTorch()
+            .WithTorchVision()
+            .WithTorchAudio()
+            .WithTorchExtraIndex("cu121")
+            .WithParsedFromRequirementsTxt(
+                await requirements.ReadAllTextAsync(cancellationToken).ConfigureAwait(false),
+                "torch"
+            );
+
+        if (installedPackage.PipOverrides != null)
+        {
+            pipArgs = pipArgs.WithUserOverrides(installedPackage.PipOverrides);
+        }
+
+        await venvRunner.PipInstall(pipArgs, onConsoleOutput).ConfigureAwait(false);
+    }
+
+    public override async Task RunPackage(
+        string installLocation,
+        InstalledPackage installedPackage,
+        RunPackageOptions options,
+        Action<ProcessOutput>? onConsoleOutput = null,
+        CancellationToken cancellationToken = default
+    )
+    {
+        await SetupVenv(installLocation).ConfigureAwait(false);
+
+        void HandleConsoleOutput(ProcessOutput s)
+        {
+            onConsoleOutput?.Invoke(s);
+
+            if (s.Text.Contains("Running on local URL", StringComparison.OrdinalIgnoreCase))
+            {
+                var regex = new Regex(@"(https?:\/\/)([^:\s]+):(\d+)");
+                var match = regex.Match(s.Text);
+
+                if (match.Success)
+                {
+                    Console.WriteLine($"AAAAAA {match.Value}");
```

Comment:
This probably isn't necessary anymore? 😅

Reply:
Oh my god. Sorry about that. Sleepy brain thought it was a good idea to test my URL condition that way. Ill remove it

Reply:
Haha no worries, thanks.